velox/vector/ComplexVector.h

```diff
@@ -722,6 +722,39 @@ class MapVector : public ArrayVectorBase {
   std::shared_ptr<MapVector> update(
       const folly::Range<DecodedVector*>& others) const;
 
+  /// Retrieves the specified keys and their associated values from each row in
+  /// a MapVector, returning a RowVector where each field corresponds to a key
+  /// and contains its respective value.
+  ///
+  /// For each row in the MapVector, the function extracts the values associated
+  /// with the specified keys and assigns them to the corresponding fields in
+  /// the output RowVector. If the map is null, a key is missing, or the value
+  /// is null, the output field will also be null. If 'replaceNulls' is set to
+  /// true, these nulls are replaced with default values appropriate for the
+  /// value type.
+  ///
+  /// Note:
+  /// - 'replaceNulls' cannot be used when the value type is ROW.
+  /// - Only keys of types SMALLINT, INTEGER, and BIGINT are supported.
+  ///
+  /// @param mapVector         The input MapVector containing rows of key-value
+  ///                          pairs.
+  /// @param keysToProject     The list of keys to extract from each map.
+  /// @param outputFieldNames  The names to assign to the corresponding fields
+  ///                          in the output RowVector.
+  /// @param replaceNulls      If true, replaces null output values with
+  ///                          type-specific defaults (not supported for ROW
+  ///                          or UNKNOWN types).
+  /// @return                  A RowVector with fields for each key in
+  ///                          keysToProject, named as specified in
+  ///                          outputFieldNames.
+  static VectorPtr toRowVector(
+      const BaseVector& map,
```

Comment:
Would you clarify whether this has to be a MapVector or can be wrapped in Dict or Const?